.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,316 @@
+        run: npm ci
+
+      - name: Building source
+        run: npm run build
+
+      - name: Downloading test reports from previous runs
+        uses: dawidd6/action-download-artifact@v6
+        with:
+          branch: main
+          if_no_artifact_found: 'ignore'
+          allow_forks: false
+          name: integration-test-junit-xml-reports-.*
+          name_is_regexp: true
+          path: tmp/junit-xml-reports-downloaded
+        continue-on-error: true
+
+      - name: Determining tests splitting by timing
+        uses: r7kamura/split-tests-by-timings@v0
+        id: split-tests
+        with:
+          reports: tmp/junit-xml-reports-downloaded
+          glob: integration_tests/tests/**/*.cy.ts
+          index: ${{ matrix.ci_node_index }}
+          total: 4
+
+      - name: Running Integration tests
+        run: TEST_RUN_ARGS="--spec $(echo ${{ steps.split-tests.outputs.paths }} | sed -E 's/ /,/g')" npm run test:integration
+
+      - name: Fix Cypress JUnit XML reports
+        run: script/fix-cypress-junit-xml
+
+      - name: Store Integration tests reports
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-test-junit-xml-reports-${{ matrix.ci_node_index }}
+          path: test_results/cypress
+
+      - name: Store Integration tests videos
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-tests-videos-${{ matrix.ci_node_index }}
+          path: integration_tests/videos
+
+      - name: Store Integration tests screenshots
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-tests-screenshots-${{ matrix.ci_node_index }}
+          path: integration_tests/screenshots
+
+  build:
+    name: Build docker image from hmpps-github-actions
+    needs:
+      - node_build
+      - linting
+      - helm_lint
+      - unit_test
+      - integration_test
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/docker_build.yml@v1
+    with:
+      docker_registry: ${{ inputs.docker_registry || 'ghcr.io' }}
+      registry_org: ${{ inputs.registry_org || 'ministryofjustice' }}
+      additional_docker_tag: ${{ inputs.additional_docker_tag }}
+      push: ${{ inputs.push || true }}
+      docker_multiplatform: ${{ inputs.docker_multiplatform || false }}
+
+  deploy_test:
+    name: Deploy to the test environment
+    needs:
+      - build
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/feature-test/') || startsWith(github.head_ref, 'feature-test/')
+    uses: ministryofjustice/hmpps-github-actions/.github/workflows/deploy_env.yml@v1
+    secrets: inherit
+    with:
+      environment: 'test'
+      app_version: '${{ needs.build.outputs.app_version }}'
+  
+  deploy_dev:
+    name: Deploy to the development environment
+    needs:
+      - build
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/feature-dev-') || startsWith(github.head_ref, 'feature-dev-')
```

Comment:
Sorry - this was wrong in my branch, could we change these to `feature-dev/` to align with the test one?